pkg/kubelet/kubelet.go

```diff
@@ -920,10 +926,20 @@ func (kl *Kubelet) listPodsFromDisk() ([]types.UID, error) {
 // Starts garbage collection threads.
 func (kl *Kubelet) StartGarbageCollection() {
 	go wait.Until(func() {
-		if err := kl.containerGC.GarbageCollect(kl.sourcesReady.AllReady()); err != nil {
-			glog.Errorf("Container garbage collection failed: %v", err)
+		kl.gcController.containerGCTrigger = make(chan struct{}, 1)
+		go wait.Until(func() {
+			kl.triggerContainerGarbageCollection()
```

Comment:
This seems a little bit strange to me. Usually we just add a timer/ticker in the `select` below (line 935) for periodic tasks. No need to create a goroutine to trigger periodically.